### DISCUSSION TITLE
Difference between `transmission-qt` and `transmission-gtk` on how they terminate

### DISCUSSION THREAD

**Discussion by stefanos82** [transmission/transmission#6261] [Q&A] [answered]
I have noticed the following behavior: when I download something via `transmission-qt` and decide to close it, it closes immediately without letting me know it closed all opened connections.

If I repeat the same procedure but with `transmission-gtk`, I can see the message "Closing connections", wait a bit to finish with opened connections, and then it terminates the app.

Is this an expected behavior by `Qt` framework?

**Comment by mikedld** [comment, marked as answer]
The framework itself is not to blame here. It's just a feature parity issue in the Qt client (so, a potential improvement that's totally possible to implement), additionally contributing to e.g. #287; for the reference, GTK client implements what you're seeing in https://github.com/transmission/transmission/blob/966756ed4b4dd8e27227516096bb4e04f6ba3603/gtk/Application.cc#L980 (so again, not a framework feature)
